otelcol/config.go

```diff
@@ -101,6 +101,20 @@ func (cfg *Config) Validate() error {
 		}
 	}
 
+	// Validate the connector configuration.
+	for connID, connCfg := range cfg.Connectors {
+		if err := component.ValidateConfig(connCfg); err != nil {
+			return fmt.Errorf("connectors::%s: %w", connID, err)
+		}
+
+		if _, ok := cfg.Exporters[connID]; ok {
+			return fmt.Errorf("connectors::%s: cannot have same id as exporter", connID)
```

Comment:
Putting myself in the shoes of someone who's looking at this message for the first time, I think the version below might be easier to grasp what's going on.

```suggestion
			return fmt.Errorf("connectors::%s: there's already an exporter named %q", connID, connID)
```